### PR TITLE
feat(scheduler): add configuration file

### PR DESCRIPTION
**Pull request by iverly**
This MR adds the configuration file for the scheduler.

The configuration file is created in the same folder where the binary is run and the file name is `scheduler.conf` in TOML format.

### CODE REVIEW

**Review by alexis-langlet — APPROVED** [review]

File: scheduler/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/// `Config` is a struct that contains the configuration of the scheduler.
+///
+/// Properties:
+///
+/// * `grpc`: This is the configuration for the gRPC server.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Config {
+    pub grpc: GrpcConfig,
+}
```
Comment (sameo) [inline scheduler/src/config.rs:11]: Why isn't this:

```rust
pub struct Config {
    pub host: String,
    pub port: u16,
}
```

File: scheduler/src/main.rs [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     env_logger::init();
     info!("starting up");
 
-    let manager = Manager::new();
+    info!("loading config");
+    let mut dir = env::current_exe()?; // get executable path
+    dir.pop(); // remove executable name
```
Comment (sameo) [inline scheduler/src/main.rs:13]: This is `env::current_dir()`.
Also you may want in a future PR to add support for finding the configuration file either through a dedicated environment variable or a CLI option for the scheduler binary.

File: scheduler/src/manager.rs [hunk 3]
```diff
@@ -68,7 +65,9 @@ impl Manager {
     /// A JoinHandle<()>
     fn create_grpc_server(&self, tx: mpsc::Sender<Event>) -> JoinHandle<()> {
         info!("creating grpc server ...");
-        let addr = "127.0.0.1:50051".parse().unwrap();
+        let addr = format!("{}:{}", self.config.grpc.host, self.config.grpc.port)
+            .parse()
+            .unwrap();
```
Comment (sameo) [inline scheduler/src/manager.rs:70]: No unwrap, please. `create_grpc_server`should return a `Result<JoinHandle<()>>`

File: scheduler/src/config.rs (hunk 1 above)
Comment (iverly) [reply scheduler/src/config.rs]: Done.

File: scheduler/src/main.rs (hunk 2 above)
Comment (iverly) [reply scheduler/src/main.rs]: Yes, that would be a nice feature.

File: scheduler/src/manager.rs (hunk 3 above)
Comment (iverly) [reply scheduler/src/manager.rs]: Fix.

**Review by sameo — APPROVED** [review]

**Pull request merged**